io.openems.edge.controller.ess.timeofusetariff/src/io/openems/edge/controller/ess/timeofusetariff/Schedule.java

```diff
@@ -0,0 +1,467 @@
+package io.openems.edge.controller.ess.timeofusetariff;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import io.openems.edge.common.type.TypeUtils;
+
+public class Schedule {
```

Comment:
I like the implementation of this class. It's clear and concise! 👍

I did not check the entire logic, but I assume both of you checked it - right?